libntfs-3g/unix_io.c

```diff
@@ -260,7 +265,14 @@ static s64 ntfs_device_unix_io_seek(struct ntfs_device *dev, s64 offset,
 static s64 ntfs_device_unix_io_read(struct ntfs_device *dev, void *buf,
 		s64 count)
 {
-	return read(DEV_FD(dev), buf, count);
+	int cnt;
+
+	if ((cnt = read(DEV_FD(dev), buf, count)) < 0) {
```

Comment:
Are there any improvements with this fix? Because of this patch, isn't the previous hyunchul's patch unnecessary?